File: 110-balanced-binary-tree/balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int check(TreeNode root,int[] maxi){
        if(root==null) return 0;

        int lh=0,rh=0;
        if(root.left!=null) lh=check(root.left,maxi);
        if(root.right!=null) rh=check(root.right,maxi);
         
        if(Math.abs(lh-rh)>1) maxi[0]=0;
        //maxi[0]=Math.max(maxi[0],lh+rh);
        return 1+Math.max(lh,rh);
    }
    public boolean isBalanced(TreeNode root) {
        int[] maxi=new int[1];
        maxi[0]=1;
        int k=check(root,maxi);
        if(maxi[0]==0) return false;
        return true;
    }
}

File: 110-balanced-binary-tree/balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        def check(node):
            if node == None:
                return 0
            
            lh = check(node.left)
            rh = check(node.right)
            if lh == -1 or rh == -1:
                return -1
            
            if abs(lh-rh) > 1:
                return -1

            return 1 + max(lh, rh)
        k = check(root)
        if k == -1:
            return False
        return True
        